### DISCUSSION TITLE
Way to get (limited) data without developer account

### DISCUSSION THREAD

**Discussion by toomeenoo** [BlazejosP/huawei-sun2000-API-CLI#16] [Show and tell]
Hi,
I've found an unofficial way to get few data for your powerplant, even if you don't have FusionSolar developer account.
You can implement it, or just mention somewhere so the others will know 😄 

First thing you have to do is login to (new) web interface of fusionsolar at [Fusion solar EU5](https://eu5.fusionsolar.huawei.com/) or your local fusionsolar server.

Then generate the KIOSK link as shown below:
![enable_kiosk](https://user-images.githubusercontent.com/32091513/119552918-59da6e80-bd9b-11eb-8403-471d4d380c2b.jpg)

The link will look like:  
~~`https://eu5.fusionsolar.huawei.com/rest/pvms/web/kiosk/v1/station-kiosk-file?kk=S0M3H45H_H3R3`~~
`https://region02eu5.fusionsolar.huawei.com/pvmswebsite/nologin/assets/build/index.html#/kiosk?kk=S0M3H45H_H3R3`

This will show you pretty web interface with some basic info.  

But what is more important with hash (HTTP URL GET parameter `kk`) cou can create:  
~~`https://eu5.fusionsolar.huawei.com/rest/pvms/web/kiosk/v1/station-kiosk-file?kk=S0M3H45H_H3R3`~~
`https://region02eu5.fusionsolar.huawei.com/rest/pvms/web/kiosk/v1/station-kiosk-file?kk=S0M3H45H_H3R3`

This link will return json response:
~~~json
{"data":"{&quot;powe...  ellipsized  ...ot;}}","success":true,"failCode":0}
~~~

In  which `data` part is again json, just html encoded:  (replace `&quot;` with `"`)  
~~~json
{
    "data": {
        "powerCurve": {
            "xAxis": [
                "00:00",
                "00:05",
                "...etc, last day of records..."
            ],
            "currentPower": "0.00",
            "activePower": [
                "0.04",
                "-",
                "...etc, last day of records..."
            ]
        },
        "stationOverview": {
            "stationName": "your_station_name",
            "plantAddress": "your_station_location",
            "stationDn": "your_station_dn"
        },
        "language": "en-US",
        "socialContribution": {
            "co2ReductionByYear": 0.28,
            "equivalentTreePlantingByYear": 1,
            "standardCoalSavings": 0.11,
            "equivalentTreePlanting": 1,
            "co2Reduction": 0.28,
            "standardCoalSavingsByYear": 0.11
        },
        "realKpi": {
            "realTimePower": "0.0",
            "cumulativeEnergy": "0.28",
            "monthEnergy": "0.26",
            "dailyEnergy": "0.14",
            "yearEnergy": "0.28"
        }
    },
    "success": true,
    "failCode": 0
}
~~~

Hope this helps someone who just want basic data, without having to deal with developer / API account  
Also this endpoint is **not official**, it's just how kiosk view works now.

**Comment by BlazejosP** [comment]
Thanks Kiosk mode was implemented before  but after Huawei server upgrade in April/May stops to work to many changes on server. I lost hope that will be possible to implement this again all previous JOSN files disappears and  I remove code from repository. But as for now I see that is possible to made this working again (Is working correctly with my account in Postman) but of course implement from scratch JOSN is different than this previous one. Your discovery allow scrap this data thanks for mentioning this. If you are interesting in contributing to repository and write this file which will make use of that discovery I will be happy  to add you  as developer. In other case I'm determine to made this also working somewhere during June because as for now I try to clean all functions which was already made and make sure that this what is written already is working with this new server. I will resurrect previous  version of **kioskmode.sh** is not working now but may be a starting point to modifications for me or you. 

Here screenshot from previous kioskmode before April upgrade
https://github.com/BlazejosP/huawei-sun2000-API-CLI/blob/master/pictures/kioskmode.png

**Comment by BlazejosP** [comment]
File is here https://github.com/BlazejosP/huawei-sun2000-API-CLI/blob/master/kioskmode.sh
that was the last version which worked flawless before upgrades on server. Now need major overhaul because original JOSN's disappears and your discovery has totally new JOSN. But some parts of code meaby usefully for example how to extract kk signature from long URL.

**Comment by BlazejosP** [comment]
> Hope this helps someone who just want basic data, without having to deal with developer / API account Also this endpoint is **not official**, it's just how kiosk view works now.

Hello  @toomeenoo 
I d'like to say thank you for providing documentation and said that your new kioskmode is already implemented inside new update and is working. So now you can use this CLI even without login or password!

**Comment by BlazejosP** [comment]
looks like we have new links after 30th December 2021


The link will look like:
`https://region02eu5.fusionsolar.huawei.com/pvmswebsite/nologin/assets/build/index.html#/kiosk?kk=S0M3H45H_H3R3`

This will show you pretty web interface with some basic info.

But what is more important with hash (HTTP URL GET parameter `kk`) cou can create:
`https://region02eu5.fusionsolar.huawei.com/rest/pvms/web/kiosk/v1/station-kiosk-file?kk=S0M3H45H_H3R3`
